playbooks/roles/crafty-bot/tasks/main.yml

```diff
@@ -0,0 +1,67 @@
+---
+- name: Allow web ports on firewall
+  ufw:
+    rule: allow
+    port: "{{ item }}"
+    proto: tcp
+  with_items:
+    - 80
+    - 443
+
+- name: Set authorized key for GitLab CI
+  authorized_key:
+    user: ubuntu
+    state: present
+    key: "{{ CI_SSH_PUBLIC_KEY }}"
+
+- name: Create the Crafty Bot directory
+  file:
+    path: "{{ APP_PATH }}"
+    state: directory
+    owner: ubuntu
+    group: ubuntu
+    mode: 0775
+
+- name: Copy the deployment script
+  copy:
+    src: deploy.sh
+    dest: "{{ APP_PATH }}/"
+    mode: 0755
+  become_user: ubuntu
+
+- name: Install the docker-compose file
+  copy:
+    src: docker-compose.yml
+    dest: "{{ APP_PATH }}/docker-compose.yml"
+    mode: 0600
+    backup: yes
+  become_user: ubuntu
+
+- name: Install docker config file
+  template:
+    src: config.json.j2
+    dest: "/home/ubuntu/.docker/config.json"
+    mode: 0600
+  become_user: ubuntu
+
+- name: Install Docker cleanup cron job
+  cron:
+    name: "Remove unused Docker data"
+    job: "docker system prune -f"
+    special_time: daily
+  become_user: ubuntu
+
+- name: Install the environment file
+  template:
+    src: env.j2
+    dest: "{{ APP_PATH }}/.env"
+    mode: 0600
+    backup: yes
+  become_user: ubuntu
+
+- name: Start Crafty Bot with docker-compose
+  docker_service:
+    project_src: "{{ APP_PATH }}"
+    pull: yes
+    debug: yes
+  become_user: ubuntu
```

Comment:
That's a lot of duplicated code. Should we consider moving it to a common playbook, e.g. something like `docker-compose-deploy`? We can add it as a TODO.

Reply:
Created SE-3919 to mitigate this.